### ISSUE TITLE
Support @Schema(hidden = true) on @Parameter

### ISSUE THREAD

**Issue by membersound** [springdoc/springdoc-openapi#148]
It would be nice to just set `@Parameter(schema = @Schema(hidden = true))` to hide the `Schema` objects entirely.

Currently, the following still results in schema generation:
```
@PostMapping("/examplePost")
@Operation(summary = "schema example")
public Object example(@Parameter(schema = @Schema(hidden = true)) JsonNode json)) {
	
}

@GetMapping("/example")
public void test(@Parameter(schema = @Schema(hidden = true) JsonNode json) {

}

```

Results in:
```
"/example": {
  "post": {
	"requestBody":{
	   "content":{
	      "*/*":{
		 "schema":{
		    "$ref":"#/components/schemas/JsonNode"
		 }
	      }
	   },
	   "required":false
        }
  }
}
....

   "JsonNode":{
      "type":"object",
      "properties":{
         "float":{
            "type":"boolean"
         },
         "array":{
            "type":"boolean"
         },
         "null":{
            "type":"boolean"
         },
         "number":{
            "type":"boolean"
         },
         "floatingPointNumber":{
            "type":"boolean"
         },
         "containerNode":{
            "type":"boolean"
         },
         "missingNode":{
            "type":"boolean"
         },
         "object":{
            "type":"boolean"
         },
         "nodeType":{
            "type":"string",
            "enum":[
               "ARRAY",
               "BINARY",
               "BOOLEAN",
               "MISSING",
               "NULL",
               "NUMBER",
               "OBJECT",
               "POJO",
               "STRING"
            ]
         },
         "pojo":{
            "type":"boolean"
         },
         "integralNumber":{
            "type":"boolean"
         },
         "short":{
            "type":"boolean"
         },
         "int":{
            "type":"boolean"
         },
         "long":{
            "type":"boolean"
         },
         "double":{
            "type":"boolean"
         },
         "bigDecimal":{
            "type":"boolean"
         },
         "bigInteger":{
            "type":"boolean"
         },
         "textual":{
            "type":"boolean"
         },
         "boolean":{
            "type":"boolean"
         },
         "binary":{
            "type":"boolean"
         },
         "valueNode":{
            "type":"boolean"
         }
      }
   }
```

**Comment by springdoc** [comment]
Hi, 

 To hide an object, you can just use hidden parameter on @Parameter annotation.
  ```@Parameter(hidden=true)```

**Comment by membersound** [comment]
I don't want to  hide the object! I only want to hide the schema displayed at the bottom of the swagger-ui.

**Comment by springdoc** [comment]
Hi @membersound, 

 Could you please the sample expected OpenAPI (json/yml) description of the behaviour you are expecting?

**Comment by membersound** [comment]
Why is this issue closed?

Here is a sample:
```
@RestController
public class TestServlet {
	@GetMapping("/test")
	public void test(JsonNode json) {

	}
}
```

Produces:
```
{
   "openapi":"3.0.1",
   "paths":{
      "/test":{
         "get":{
            "operationId":"test",
            "parameters":[
               {
                  "name":"json",
                  "in":"query",
                  "required":true,
                  "schema":{
                     "$ref":"#/components/schemas/JsonNode"
                  }
               }
            ],

            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      },
      "/testPost":{
         "post":{
            "operationId":"testPost",
            "requestBody":{
               "content":{
                  "*/*":{
                     "schema":{
                        "$ref":"#/components/schemas/JsonNode"
                     }
                  }
               }
            },
            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      }
   },
   "components":{
      "schemas":{
         "JsonNode":{
            "type":"object",
            "properties":{
               "float":{
                  "type":"boolean"
               },
               "array":{
                  "type":"boolean"
               },
               "null":{
                  "type":"boolean"
               },
               "number":{
                  "type":"boolean"
               },
               "floatingPointNumber":{
                  "type":"boolean"
               },
               "containerNode":{
                  "type":"boolean"
               },
               "missingNode":{
                  "type":"boolean"
               },
               "object":{
                  "type":"boolean"
               },
               "nodeType":{
                  "type":"string",
                  "enum":[
                     "ARRAY",
                     "BINARY",
                     "BOOLEAN",
                     "MISSING",
                     "NULL",
                     "NUMBER",
                     "OBJECT",
                     "POJO",
                     "STRING"
                  ]
               },
               "pojo":{
                  "type":"boolean"
               },
               "integralNumber":{
                  "type":"boolean"
               },
               "short":{
                  "type":"boolean"
               },
               "int":{
                  "type":"boolean"
               },
               "long":{
                  "type":"boolean"
               },
               "double":{
                  "type":"boolean"
               },
               "bigDecimal":{
                  "type":"boolean"
               },
               "bigInteger":{
                  "type":"boolean"
               },
               "textual":{
                  "type":"boolean"
               },
               "boolean":{
                  "type":"boolean"
               },
               "binary":{
                  "type":"boolean"
               },
               "valueNode":{
                  "type":"boolean"
               }
            }
         }
      }
   }
}
```

This generates:
![test](https://user-images.githubusercontent.com/5269630/68576516-7f7c7d00-046e-11ea-89c9-595855cef418.jpg)

I want to hide the "JsonNode" output in "Schema" down below.

**Comment by springdoc** [comment]
Hi @membersound,

Could you please the sample expected OpenAPI (json/yml) description of the behaviour you are expecting?

 Your resuest do not seem relevant, as if we hide thé generated schema related, the OpenAPI description will be invalid.
 But if you just clarify your expected result (using OpenApi yaml/json), we can have better understanding of your request.

**Comment by membersound** [comment]
If I want to hide the `schema` of a parameter, I would expect both the schema node in the `requestBody   and also in the `components` section being empty:

```
{
   "openapi":"3.0.1",
   "paths":{
      "/test":{
         "get":{
            "operationId":"test",
            "parameters":[
               {
                  "name":"json",
                  "in":"query",
                  "required":true,
                  "schema":{
                     "$ref":"#/components/schemas/JsonNode"
                  }
               }
            ],

            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      },
      "/testPost":{
         "post":{
            "operationId":"testPost",
            "requestBody":{
               "content":{
                  "*/*":{
                     "schema":{ }
                  }
               }
            },
            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      }
   },
   "components":{
      "schemas": { }
   }
}
```

**Comment by springdoc** [comment]
Hi, 

 Please copy and past your OpenAPI description on https://editor.swagger.io/, and you will see that the expected OpenAPI description is invalid: 

`Semanticerror at paths./test.get.parameters.0.schema.$ref
$refs must reference a valid location in the document
Jump to line 11` 

 This part of the OpenAPI Descprition that you expect "$ref":"#/components/schemas/JsonNode", can never be genereted using @Schema(hidden = true) annotation.

 If you want to change the generated schema, you can use $ref attribute instead. (This supposes you have declared the type JsonNode in a previous section of your code, to have a correct OpenApi Description)
```java
    @RestController
    public class TestServlet {
        @GetMapping("/test")
        public void test(@Parameter(ref = "#/components/schemas/JsonNode") JsonNode json) {

        }
    }
` ` `

**Comment by membersound** [comment]
Sorry, I forgot to update the GET endpoint. Here is the example I'd expect:

```
{
   "openapi":"3.0.1",
   "info": {
    "title": "example",
    "version": "1.0.0"
   },
   "paths":{
      "/test":{
         "get":{
            "operationId":"test",
            "parameters":[
               {
                  "name":"json",
                  "in":"query",
                  "required":true,
                  "schema":{ }
               }
            ],

            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      },
      "/testPost":{
         "post":{
            "operationId":"testPost",
            "requestBody":{
               "content":{
                  "*/*":{
                     "schema":{ }
                  }
               }
            },
            "responses":{
               "200":{
                  "description":"default response"
               }
            }
         }
      }
   },
   "components":{
      "schemas": { }
   }
}
```

**Comment by springdoc** [comment]
Hi, 

 The fix will be available on v1.2.7, of springdoc-openapi.
